Add introduction vignette (#10)
epiverse-trace/superspreading: vignettes/superspreading.Rmd

:::

```{r setup}
library(superspreading)
library(epiparameter)
library(distributional)
```

## Probability of epidemic

The probability that a novel disease will cause a epidemic (i.e. sustained 
transmission in the population) is determined by the nature of that diseases'
transmission heterogeneity. This variability may be an intrinsic property of 
the disease, or a product of human behaviour and social mixing patterns.

For a given value of $R$ (or $R_0$), that is the number of secondary cases caused by a 
primary case, if the variability is high, the probability that the outbreak 
will cause epidemic is lower as the superspreading events are rare. Whereas for
lower variability the probability is higher as more individuals are closer to the 
mean ($R$). The `probability_epidemic()` function in {superspreading} can calculate
this probability. `k` is the dispersion parameter of a negative binomial distribution
and controls the variability of individual-level transmission.

```{r, prob-epidemic-k}
probability_epidemic(R = 1.5, k = 1, a = 1)
probability_epidemic(R = 1.5, k = 0.5, a = 1)
probability_epidemic(R = 1.5, k = 0.1, a = 1)
```

> In the above code values above to one are low heterogeneity (in the case `k` 
> &rarr; $\infty$ it is a poisson distribution), and as it decreases to zero 
> heterogeneity increases. When `k` equals 1 the distribution geometric. Values
> of `k` less than one indicate overdispersion of disease tranmsission, a signature
> of superspreading.

Reviewer: ```suggestion
In the above code, `k` values above one represent low heterogeneity (in the case `k` 
&rarr; $\infty$ it is a poisson distribution), and as `k` decreases,
heterogeneity increases. When `k` equals 1, the distribution is geometric. Values
of `k` less than one indicate overdispersion of disease tranmsission, a signature
of superspreading.
```

probability_epidemic(R = 5, k = 1, a = 1)
```

Any value of $R$ less than or equal to one will have zero probability of causing
a sustained epidemic.

Lastly on the probability of transmission dynamics causing an epidemic, the 
number of initial infections seeding the outbreak will influence the probability;
`a` is the number of initial infections.

```{r, prob-epidemic-a}
probability_epidemic(R = 1.5, k = 1, a = 1)
probability_epidemic(R = 1.5, k = 1, a = 10)
probability_epidemic(R = 1.5, k = 1, a = 100)
```

## Empirical superspreading

Given `probability_epidemic()` it is possible to determine the probability of 
an epidemic for diseases for which parameters of an offspring distribution have 
been estimated. An offspring distribution is simply the distribution of the 
number of secondary infections caused by a primary infection. It is the distribution
of $R$, with the mean of the distribution given as $R$.

Here we can use {epiparameter} to load in offspring distributions for multiple 
diseases and evaluate how likely they are to cause epidemics.

```{r, epiparam}
sars <- epidist_db(
  disease = "SARS",
  epi_dist = "offspring_distribution"
)

> evd <- epidist_db(

Reviewer: This returned a `parameters class not recognised` error, although `evd` still defined as an object.

Author: This will be due to an older version of {epiparameter}, it runs correctly on the github actions which installs the latest version. If this problem persists please raise it as an issue.

  disease = "Ebola Virus Disease",
  epi_dist = "offspring_distribution"
)
```

The parameters of each distribution can be extracted:

```{r, params}
sars_params <- parameters(sars)
sars_params
evd_params <- parameters(evd)
evd_params
```

```{r, prob-epidemic-empiric}
family(sars)
probability_epidemic(
  R = sars_params[["mean"]],
  k = sars_params[["dispersion"]],
  a = 1
)
family(evd)
# k is set to infinite as the distribution for EVD is poisson

> probability_epidemic(R = evd_params[["mean"]], k = Inf, a = 1)

Reviewer: Should this be `evd_params[["l"]]`? Also, it looks like there may be a bug with `Inf`, as `probability_epidemic(R = evd_params[["l"]], k = 1e6, a = 1)` returns 0.583 (which is more plausible than the 0.001 returned with `Inf` – which looks like bottom end of grid search)

Author: The `evd_param[["l"]]` will be due to an old version of {epiparameter}. 

Thanks for raising the issue around the use of `Inf` in `probability_epidemic()`. I'll look into this.

Author: Resolved in cb7e47f37bcc5618176f4927e29fe62e7864e01a by changing infinite values of `k` to `1e10` in `probability_epidemic()`.

```

In the above example we assume the initial pool of infectors is one (`a = 1`) 
but this can easily be adjusted in the case there is evidence for a larger initial

> seeding of infections.

Reviewer: ```suggestion
seeding of infections, whether from animal-to-human spillover or imported cases from outside the area of interest.
```

We can see that the probability of an epidemic given the estimates of 
@lloyd-smithSuperspreadingEffectIndividual2005 is greater for SARS than Ebola. 
This is due to the offspring distribution of Ebola being poisson distributed 
with a smaller mean (mean $R$ = `r evd_params[["mean"]]`), 
compared to SARS, which is negative binomially distributed with a relatively
small dispersion (k = `r sars_params[["dispersion"]]`).

> ## References

Reviewer: Could cite this one for COVID: https://www.thelancet.com/article/S1473-3099(20)30144-4/fulltext
